Default workers inside docker image

SysOfM opened the pull request:
Changed the build of the Docker image, so that the image was minimal in size and contained ready-made default workers inside.
For better Dockerfile code changed the server:Makefile. All functionality of the Makefile build is saved.
Right now these solutions work in our Kubernetes cluster.

SysOfM committed:
Changes in Dockerfile:
- Code refactored
- Built steps to create less artifacts
- Updated erlang image and kubectl tool
- Build and installation of workers as a default scenario

Added .dockerignore

SysOfM committed:
- Made the build workers as tgz packages exactly as the author did
- Source code should not be deleted, it will be needed when deploying. Just need to clean it well.
- Prepared the archives of nodes and configured paths for their rapid deployment
- Did the correct configuration of server.config
- Also we neet to run with `{auto_update_deployed_code, disable}, {custom_os_code_builds, disable}` for fast node deployment
- Fixed the removal of the archive with kubectl
- Changed the CMD and COPY source code

SysOfM committed:
- Prepare image so that the workers have already been established
- Changed config installation to node_deployment_path and removed .git

SysOfM committed:
Changed the server Makefile to build workers from Dockerfile

SysOfM committed:
Made a multi-stage docker build to save image size

SysOfM committed:
Remioved MZBENCH_SRC_DIR

SysOfM committed:
Code style fix

SysOfM committed:
Removed extra space

parsifal-47 commented:
I'm not sure I understand everything here, it is more than just about adding workers I suppose :)

but anyway, two things:
1. please resubmit this one to mzbench/mzbench
2. @gliush as a member you have right to make decision

feel free to ask questions
Thanks!

SysOfM commented:
> I'm not sure I understand everything here, it is more than just about adding workers I suppose :)

Yeah, you're right. 
I had to remake the Dockerfile so that the image did not contain unnecessary files.

> 1. please resubmit this one to mzbench/mzbench

Ok, I'll do it

SysOfM committed:
Pip requirements needs to run workers

SysOfM committed:
Link for node package with git version

SysOfM committed:
Hardcoded fix before all code will be written

SysOfM committed:
Fixed a bug, were k8s workers not installed.
Added `provision_workers` option to install workers separately from `provision_nodes` option.

SysOfM committed:
Increased version to 0.5.6
